### PR TITLE
reupload of assignment 4

### PR DESCRIPTION
**Pull request by NestorRovira**

### CODE REVIEW

File: Assignment4/nestor_rovira-22c057/Task07.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+# **TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**
+
+# In[4]:
+
+
+q1 = """select distinct ?sub 
+where
+{
+?sub rdfs:subClassOf ns:LivingThing.
```
Comment (dgarijo) [inline Assignment4/nestor_rovira-22c057/Task07.py:40]: this does not retrieve subclasses of subclasses of living thing

File: Assignment4/nestor_rovira-22c057/Task07.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+# In[5]:
+
+
+q2 = """select distinct ?individuals 
+where
+{
+{?individuals a ns:Person.}
+union
+{?sub rdfs:subClassOf ns:Person. ?individuals a ?sub}
```
Comment (dgarijo) [inline Assignment4/nestor_rovira-22c057/Task07.py:59]: same as above

File: Assignment4/nestor_rovira-22c057/Task07.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+# In[6]:
+
+
+q3 = """select distinct ?individuals ?property
+where
+{
+?sub rdfs:subClassOf ns:LivingThing.
+?individuals a ?sub.
+?individuals ?property ?value.
```
Comment (dgarijo) [inline Assignment4/nestor_rovira-22c057/Task07.py:78]: this is not doing what the exercise asks for

File: Assignment4/nestor_rovira-22c057/Task07.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+# **TASK 7.4:  List the name of the persons who know Rocky**
+
+# In[9]:
+
+
+q4 = """select ?individuals
+where
+{
+?individuals foaf:knows ns:RockySmith.
```
Comment (dgarijo) [inline Assignment4/nestor_rovira-22c057/Task07.py:94]: the exercise is asking for names, not uris

File: Assignment4/nestor_rovira-22c057/Task07.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+{
+?individuals foaf:knows ?x.
+?individuals foaf:knows ?y.
+filter(?x != ?y).
+}"""
+
+# Visualize the results
+for r in g.query(q5):
+  print(r)
+
```
Comment (dgarijo) [inline Assignment4/nestor_rovira-22c057/Task07.py:118]: missing all exercises in rdflib

**Review by dgarijo — CHANGES_REQUESTED** [review]
Please address the remaining comments

**Comment by anaigmo** [comment]
Entregado sin corregir los comentarios, no se admiten más PR

**Pull request merged**